cli/js/tests/unit_test_runner.ts

```diff
@@ -47,62 +182,63 @@ async function main(): Promise<void> {
     console.log("\t" + fmtPerms(perms));
   }
 
-  const testResults = new Set<TestResult>();
+  const testResults = new Set<PermissionSetTestResult>();
+  const consoleReporter = new Deno.ConsoleTestReporter();
 
   for (const perms of permissionCombinations.values()) {
-    const permsFmt = fmtPerms(perms);
-    console.log(`Running tests for: ${permsFmt}`);
-    const cliPerms = permsToCliFlags(perms);
-    // run subsequent tests using same deno executable
-    const args = [
-      Deno.execPath(),
-      "run",
-      ...cliPerms,
-      "cli/js/tests/unit_tests.ts"
-    ];
-
-    const p = Deno.run({
-      args,
-      stdout: "piped"
-    });
-
-    const { actual, expected, resultOutput } = await parseUnitTestOutput(
-      p.stdout!,
-      true
-    );
-
-    let result = 0;
-
-    if (!actual && !expected) {
-      console.error("Bad cli/js/tests/unit_test.ts output");
-      result = 1;
-    } else if (expected !== actual) {
-      result = 1;
-    }
-
-    testResults.add({
-      perms: permsFmt,
-      output: resultOutput,
-      result
-    });
+    const result = await runTestsForPermissionSet(consoleReporter, perms);
+    testResults.add(result);
   }
 
   // if any run tests returned non-zero status then whole test
   // run should fail
-  let testsFailed = false;
+  let testsPassed = true;
 
   for (const testResult of testResults) {
-    console.log(`Summary for ${testResult.perms}`);
-    console.log(testResult.output + "\n");
-    testsFailed = testsFailed || Boolean(testResult.result);
+    const { permsStr, stats, duration } = testResult;
+    console.log(`Summary for ${permsStr}`);
+    await consoleReporter.end({
+      kind: Deno.TestEvent.End,
+      stats,
+      duration,
+      results: []
+    });
+    testsPassed = testsPassed && testResult.passed;
   }
 
-  if (testsFailed) {
+  if (!testsPassed) {
     console.error("Unit tests failed");
     Deno.exit(1);
   }
 
   console.log("Unit tests passed");
 }
 
+async function main(): Promise<void> {
+  // TODO: use `flags` parser
+  const args = Deno.args;
+
+  const isWorker = args.includes("--worker");
+
+  if (!isWorker) {
+    return await masterRunnerMain();
+  }
+
+  const addrArg = args.find(e => e.includes("--addr"));
+  assert(typeof addrArg === "string");
+  const addrStr = addrArg.split("=")[1];
+  const [hostname, port] = addrStr.split(":");
+  const addr = { hostname, port: Number(port) };
+
+  let perms: Deno.PermissionName[] = [];
+  const permsArg = args.find(e => e.includes("--perms"));
+  assert(typeof permsArg === "string");
+  const permsStr = permsArg.split("=")[1];
+  if (permsStr.length > 0) {
+    perms = permsStr.split(",") as Deno.PermissionName[];
+  }
+
+  await workerRunnerMain(addr, perms);
+}
+
```

Comment:
Should this file be split into two separate entry points?

Reply:
This works for me

Reply:
Done